src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildANetCoreApp.cs

```diff
@@ -925,5 +926,107 @@ public void It_builds_the_project_successfully_with_only_reference_assembly_set(
+        {
+            var packageProject = new TestProject()
+            {
+                Name = "WithRidAssets",
+                TargetFrameworks = targetFramework,
+            };
+
+            // Add assets for each RID. The test just needs the asset to exist, so it can just copy same output assembly.
+            foreach (string rid in rids)
+            {
+                if (addLibAssets)
+                {
+                    packageProject.AddItem("None",
+                        new Dictionary<string, string>()
+                        {
+                        { "Include", $"$(TargetPath)" },
+                        { "Pack", "true" },
+                        { "PackagePath", $@"runtimes\{rid}\lib\$(TargetFramework)" }
+                        });
+                }
+
+                if (addNativeAssets)
+                {
+                    packageProject.AddItem("None",
+                        new Dictionary<string, string>()
+                        {
+                            { "Include", $"$(TargetPath)" },
+                            { "Pack", "true" },
+                            { "PackagePath", $@"runtimes\{rid}\native" }
+                        });
+                }
+            }
+
+            // Identifer based on test inputs to create test assets that are unique for each test case
+            string assetIdentifier = $"{nameof(It_warns_on_nonportable_rids)}{targetFramework}{string.Join(null, rids)}{addLibAssets}{addNativeAssets}{useRidGraph}{shouldWarn}";
+
+            var packCommand = new PackCommand(_testAssetsManager.CreateTestProject(packageProject, assetIdentifier));
+            packCommand.Execute().Should().Pass();
+            var package = new TestPackageReference(packageProject.Name, "1.0.0", packCommand.GetNuGetPackage());
+
+            var testProject = new TestProject()
+            {
+                Name = "NonPortableRid",
+                TargetFrameworks = targetFramework,
+                IsExe = true
+            };
+
+            // Reference the package, add it to restore sources, and use a test-specific packages folder 
+            testProject.PackageReferences.Add(package);
+            testProject.AdditionalProperties["RestoreAdditionalProjectSources"] = Path.GetDirectoryName(package.NupkgPath);
+            testProject.AdditionalProperties["RestorePackagesPath"] = @"$(MSBuildProjectDirectory)\packages";
+
+            // The actual list comes from BundledVersions.props. For testing, we conditionally add a
+            // subset of the list if it isn't already defined (so running on an older version)
+            testProject.AddItem("_KnownRuntimeIdentiferPlatforms",
+                new Dictionary<string, string>()
+                {
+                    { "Include", "unix" },
+                    { "Condition", "'@(_KnownRuntimeIdentiferPlatforms)'==''" }
+                });
```

Comment:
It's probably a good idea to go back and delete this once the installer changes have been merged and flowed into stage 0.

Reply:
Created https://github.com/dotnet/sdk/issues/33152 so I don't forget this.